contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_encoder_psnr.py

```diff
@@ -0,0 +1,525 @@
+        """
+        super().__init__(**kwargs)
+        self._codec_parser_map = {
+            "v4l2h264enc": "h264parse",
+            "v4l2h265enc": "h265parse",
+        }
+        self._base_pipeline = (
+            "{} filesrc location={} ! decodebin ! videoconvert !"
+            " video/x-raw,format={} ! {}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._color_space,
+            self._codec,
+        )
+
+    def _264_265_command_builder(self) -> List[str]:
+        base_pipeline = (
+            "{} filesrc location={} ! decodebin ! videoconvert !"
+            " video/x-raw,format={} ! {}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._color_space,
+            self._codec,
+        )
+
+        if self._mux in [MuxType.MP4.value, MuxType.MATROSKA.value]:
+            encode_parser = self._codec_parser_map.get(self._codec)
+            final_pipeline = "{} ! {} ! {} ! filesink location={}".format(
+                base_pipeline,
+                encode_parser,
+                self._mux,
+                self._output_file_full_path,
+            )
+        elif self._mux == MuxType.AVI.value:
+            final_pipeline = "{} ! {} ! filesink location={}".format(
+                base_pipeline, self._mux, self._output_file_full_path
+            )
+        else:
+            raise SystemExit(
+                "Error: Pipeline for '{}' mux not implemented.".format(
+                    self._mux
+                )
+            )
+
+        return final_pipeline
+
+    def _v4l2jpecenc_command_builder(self) -> List[str]:
+        if self._platform == "genio-350":
+            raise SystemExit(
+                "Genio 350 platform doesn't support v4l2jpecenc codec"
+            )
+        # Make the name of artifact and its full path
+        self._output_file_name = "encode_psnr_{}x{}_{}.jpg".format(
+            self._width, self._height, self._color_space
+        )
+        self._output_file_full_path = os.path.join(
+            PLAINBOX_SESSION_SHARE, self._output_file_name
+        )
+        # Capture the first frame and save it as jpg file
+        final_pipeline = (
+            "{} filesrc location={} ! decodebin ! videorate !"
+            " video/x-raw,framerate=1/1 ! videoconvert ! "
+            "video/x-raw,format={} ! {} ! filesink location={}"
+        ).format(
+            GST_LAUNCH_BIN,
+            self._golden_sample,
+            self._color_space,
+            self._codec,
+            self._output_file_full_path,
+        )
+        # Reassign the reference file to be an image file
+        self._psnr_reference_file = os.path.join(
+            VIDEO_CODEC_TESTING_DATA,
+            SAMPLE_2_FOLDER,
+            "big_bug_bunny_{}x{}.jpg".format(self._width, self._height),
+        )
+        return final_pipeline
+
+    def _build_command(self) -> str:
```

Comment:
I think this method returns a `list[str]` not `str`

Reply:
See https://github.com/canonical/checkbox/pull/1491/commits/18f418371c7f3696951ffcb95a1731d385bd066b